### PR TITLE
Doc - fix definition of real_sph_harm functions

### PR DESCRIPTION
**Pull request by MrBago**
Can someone double check me, but I think this is right now.

### CODE REVIEW

File: dipy/reconst/shm.py [hunk 1]
```diff
@@ -203,9 +203,9 @@ def real_sph_harm(m, n, theta, phi):
 
     Where the real harmonic $Y^m_n$ is defined to be:
 
-        Real($Y^m_n$) * sqrt(2) if m > 0
-        $Y^m_n$                 if m == 0
-        Imag($Y^m_n$) * sqrt(2) if m < 0
+        Imag($Y^m_n$) * sqrt(2)     if m > 0
+        $Y^m_n$                     if m == 0
```
Comment (arokem) [inline dipy/reconst/shm.py]: I think that's correct. I believe that we are following the same convention as the one outlined in @mdesco et al.'s 2007 paper (http://onlinelibrary.wiley.com/doi/10.1002/mrm.21277/full). See equation 3. I believe you can write this central line as: 

    $Y^0_n$     if m=0 

I don't think that we need to collapse the index here, as they do in that paper.

**Comment by arokem** [comment]
Any more comments here? I'll merge this in a couple of days, if I don't hear from anyone.

File: dipy/reconst/shm.py (hunk 1 above)
Comment (MrBago) [inline dipy/reconst/shm.py]: I changed the middle line, does this look better?

**Comment by arokem** [comment]
LGTM. Merging.

**Pull request merged**